daliuge-engine/dlg/apps/pyfunc.py

```diff
@@ -455,6 +468,11 @@ def run(self):
                         if ptype in ["Complex", "Json"]:
                             try:
                                 value = ast.literal_eval(value)
+                            except ValueError:
+                                pass
+                        elif ptype in ["Python"]:
+                            try:
+                                value = eval(value)
```

Comment:
Not a blocking issue, but it would be great if we could restrict the modules/builtins available for use in eval() via the additional globals/locals arguments.

https://docs.python.org/3/library/functions.html#eval

I'm sure it requires some thought though, since we don't want to restrict users too much.